src/stack_pr/cli.py

```diff
@@ -263,32 +272,30 @@ def head(self) -> str:
         return self._head
 
     @head.setter
-    def head(self, head: str):
+    def head(self, head: str) -> None:
         self._head = head
 
     def has_head(self) -> bool:
         return self._head is not None
 
     @property
-    def base(self) -> str:
-        if self._base is None:
-            raise ValueError("base is not set")
+    def base(self) -> str | None:
         return self._base
 
     @base.setter
-    def base(self, base: str):
+    def base(self, base: str | None) -> None:
```

Comment:
I believe `str | None` is a 3.10 feature, while we try to support 3.9, so I'd recommend the older syntax `Optional[str]` here.

Reply:
github code review is wrong :) if you do `from __future__ import annotations` you can do the modern syntax all the way back to python 3.7+.